### ISSUE TITLE
Question: Why does 02 get prepended to the public key?

### ISSUE THREAD

**Issue by chmac** [nbd-wtf/nostr-tools#82]
I'm reading this part of the nip04 code:

https://github.com/nbd-wtf/nostr-tools/blob/8b18341ebb5a0ee45176ce45351872f1b7e23c63/nip04.ts#L12

The format seems to be the same for encryption and decryption.

Please forgive me if this is a stupid question, but why is there a `02` prepended to the public key here?

I've looked at the [noble docs](https://github.com/paulmillr/noble-secp256k1#getsharedsecretprivatekeya-publickeyb) and I can't figure it out from there either.

**Comment by fiatjaf** [comment]
Public keys are points in a curve.

ECDSA public keys are normally hex-encoded with `("02" or "03") + x_coordinates`. For each `x` coordinates there are two possible points. The `02` or `03` exist to determine that.

BIP340 public keys, and therefore Nostr public keys, however, are encoded without that initial prefix, and, when necessary, by convention, they are assumed to have the `02` (which means that the `y` is even and not odd).

**Comment by chmac** [comment]
@fiatjaf Awesome, thanks for the super fast reply and the explanation. Is this documented somewhere? Could I submit a PR somewhere to add the info you wrote here? Maybe this stuff is all super obvious to folks more familiar with cryptography. Feel free to ignore this comment if adding this doesn't make sense.

**Comment by fiatjaf** [comment]
Somewhere in https://bips.xyz/340.

I don't think this NIP is appropriate, but I think we should have a very Basic Cryptography Introduction for Nostr Programmers and Users somewhere.

Do you think the explanation above made sense? If yes, then I might expand it and publish somewhere. I am a beginner in these subjects too and I struggle with the basic stuff so I might just be in a good position to explain it. But I don't know.

**Comment by chmac** [comment]
@fiatjaf Definitely. I have some experience using crypto libraries but never writing them, and I thought the explanation was excellent. I planned to copy and paste it somewhere to create a PR! :-)

Personally, I would suggest that adding a note to the NIP 04 makes sense, even if just to say that the `y` is even and not odd. I guess somebody who knows how keys are encoded could extract that from the example in the NIP, but it would be clearer if it was explicitly stated somewhere in my opinion.

A broader intro to cryptography in nostr would also be great. Maybe in the main nostr repo? I realise the NIPs have moved to their own repo, but there is already an FAQ in the main repo. Adding even just 1 question with the answer you posted above to the FAQ could be a good first step?
